external/mqtt/src/test/scala/org/apache/spark/streaming/mqtt/MQTTStreamSuite.scala

```diff
@@ -17,31 +17,65 @@
 
 package org.apache.spark.streaming.mqtt
 
-import org.scalatest.FunSuite
-
+import org.scalatest.{BeforeAndAfter, FunSuite}
+import org.scalatest.concurrent.Eventually
+import scala.concurrent.duration._
 import org.apache.spark.streaming.{Seconds, StreamingContext}
 import org.apache.spark.storage.StorageLevel
 import org.apache.spark.streaming.dstream.ReceiverInputDStream
+import org.eclipse.paho.client.mqttv3._
+import org.eclipse.paho.client.mqttv3.persist.MqttDefaultFilePersistence
 
-class MQTTStreamSuite extends FunSuite {
-
-  val batchDuration = Seconds(1)
+class MQTTStreamSuite extends FunSuite with Eventually with BeforeAndAfter {
 
+  private val batchDuration = Seconds(1)
   private val master: String = "local[2]"
-
   private val framework: String = this.getClass.getSimpleName
+  private val brokerUrl = "tcp://localhost:1883"
```

Comment:
TCP/IP port 1883 is reserved with IANA for use with MQTT. That is why I hardcoded it.